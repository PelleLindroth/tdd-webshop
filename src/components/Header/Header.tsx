import { Link } from 'react-router-dom'
import { IProduct } from '../../mocks/products'

// /login: conditional if user is not logged in//}

const Header = (props: { cart: IProduct[] }) => {
  const { cart } = props

  return (
    <header>
      <h1>THIS IS THE WEBSHOP</h1>
      <h2>Items in cart: {cart ? cart.length : 0}</h2>
      <nav>
        <Link data-testid="home" to="/">
          Home
        </Link>
        <Link data-testid="login" to="/login">
          Login
        </Link>
        <Link data-testid="cart" to="/cart">
          My Cart
        </Link>
      </nav>
    </header>
  )
}

export default Header
