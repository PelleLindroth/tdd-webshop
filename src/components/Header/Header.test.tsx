import { shallow } from 'enzyme'
import Header from './Header'

describe('Header', () => {
  it('renders Header component correctly', () => {
    const wrapper = shallow(<Header />)

    expect(wrapper).toMatchSnapshot()
  })
})
