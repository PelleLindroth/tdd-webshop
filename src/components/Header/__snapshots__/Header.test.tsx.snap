// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders Header component correctly 1`] = `
<header>
  <h1>
    THIS IS THE WEBSHOP
  </h1>
  <h2>
    Items in cart: 
    0
  </h2>
  <nav>
    <Link
      data-testid="home"
      to="/"
    >
      Home
    </Link>
    <Link
      data-testid="login"
      to="/login"
    >
      Login
    </Link>
    <Link
      data-testid="cart"
      to="/cart"
    >
      My Cart
    </Link>
  </nav>
</header>
`;
